include/zpp/compiler.hpp

```diff
@@ -0,0 +1,18 @@
+/*
```

Comment:
What's the motivation for this file?

To me the name suggests it's a place where compiler-specific adjustments might be made.  The only role it has right now is to confirm that a C++ compiler is being used, and the proper version of C++ has been selected.

Would naming it `common.hpp` make more sense?  Or is there existing practice for `compiler.hpp` in other frameworks?

Also I don't think it's necessary/desirable to check for C++ version compatibility here; that can be ignored or done in the headers that use such features.  Compilers should complain when asked for features that aren't supported.  It's probably not even necessary to test for C++ compilers; inclusion by a C compiler should produce obvious diagnostics very quickly.

Reply:
The motivation was to prevent user complains due to obscure errors (C++ likes to generate obscure errors, when it doesn't understand the syntax). The name is because it checks the compiler, but for the rest it is just a name, and the user should never have to include directly.

Reply:
My point is by naming it `compiler.hpp` you suggest it's only useful for the compiler.  If you need something that's useful in every header but isn't not compiler specific, then you end up creating `common.hpp` and including that in every file too.  IMO you gain nothing by using a name that implies such a restricted role.